docs/rfcs/save-search-filter.md

```diff
@@ -0,0 +1,81 @@
+# Save Filters Summary
+
+"Save Filters" saves the filters set for each feature list in the console.
+
+This feature allows you to give default settings to saved filters.
+
+This feature should be designed to store filter information for the various information displayed in the console.
+
+## package
+packageName: savefilter
+
+## Function
+Filter and search information is based on the string in the query part of the URL.
+
+ex) https://dev.bucketeer.jp/default/features?page=1&q=test&sort=-createdAt&tagIds=iOS
+
+Filter and Search Info: ```q=test&sort=-createdAt&tagIds=iOS```
+
+### Create
+Receives and saves requests for filter titles, query strings, and default settings flags.
+* Request
+  * Name: Filter Name String
+  * Query: Query Parameters String
+  * DefaultFlag: Filter flags to set by default
+  * TargetType: Target feature type to filter (FeatureFlag, Goal, etc.)
+* Request Validation
+  * Name: Reject empty string(Same name is allowed)
+  * Query: Reject empty string(Same query is allowed)
+  * DefaultFlag: ー
+  * TargetType: Reject Unknown Type
+
+### Update
+Receives and updates requests for filter titles, query strings, and default settings flags.
+* Request
+  * ID: Filter ID
+  * Name: Filter Name String
+  * Query: Query Parameters String
+  * DefaultFlag: Filter flags to set by default
+* Request Validation
+  * ID: ID that does not exist
+  * Name: Reject empty string(Same name is allowed)
+  * Query: Reject empty string(Same name is allowed)
+  * DefaultFlag: ー
+
+### Delete
+Receives a request for the FilterID to be deleted and deletes the corresponding filter.
+* Request
+  * ID: Filter ID
+* Request Validation
+  * ID: Reject ID that does not exist
+
+### List
+A list is retrieved for each filter target(FeatureFlag, Goal, etc.).
+* Request
+  * TargetType: Target feature type to filter (FeatureFlag, Goal, etc.)
+* Request Validation
+  * TargetType: Reject Unknown Type
+
+## Implementation
+
+### Storage
+
+* Create `search_filter` table definition.
+
+```sql
+      CREATE TABLE `search_filter` (
```

Comment:
How do you plan to connect the filter to the user logged in to the console?

Reply:
I have a few questions regarding this comment.

Since this comment only has `Create`, does it mean that a `SELECT` type query is also required to obtain the search list?

Or is the intent of how it's supposed to be managed since the Create request doesn't include any information about the user?
I was thinking of saving per environment, not per user. (I just realized it's better to run it per project instead of per environment.)
I thought it would be useful to be able to share saved search results with your team.What do you think?
Or is it better to run it for each user? (It might not be that difficult if you share the URL with each user.)

Reply:
Since each user has his own preference, the feature is more useful per user not per environment or project.

Reply:
Since the filter is per user, have you considered saving the filters in the account table?